### ISSUE TITLE
jx version does gets confused between the new and the old releasse

### ISSUE THREAD

**Issue by vfarcic** [jenkins-x/jx#4115]
### Summary 

When upgrading `jx` version, it offers a new release but instead upgrading the CLI it claims that I'm already running the latest.

### Steps to reproduce the behavior

```bash
jx version
```

```
jx version
NAME               VERSION
jx                 2.0.184
jenkins x platform 2.0.330
Kubernetes cluster v1.12.8
kubectl            v1.14.2
helm client        Client: v2.14.0+g05811b8
git                git version 2.20.1 (Apple Git-117)
Operating System   Mac OS X 10.14.5 build 18F203
WARNING:
A new jx version is available: 2.0.186? Would you like to upgrade to the new jx version? Yes

You are already on the latest version of jx 2.0.184
```

### Expected behavior

It should have upgraded to 2.0.186 as suggested.

### Actual behavior

It claims that 2.0.184 is the latest version even though it offered to upgrade to 2.0.186.

### Jx version

The output of `jx version` is:

```
NAME               VERSION
jx                 2.0.184
jenkins x platform 2.0.330
Kubernetes cluster v1.12.8
kubectl            v1.14.2
helm client        Client: v2.14.0+g05811b8
git                git version 2.20.1 (Apple Git-117)
Operating System   Mac OS X 10.14.5 build 18F203
```

### Jenkins type

Any

### Kubernetes cluster

Any

### Operating system / Environment

macOS

**Comment by garethjevans** [comment]
Are you using brew to install jx?

**Comment by vfarcic** [comment]
Yes.

**Comment by garethjevans** [comment]
I think this is caused by the jx check looking directly at github, but the homebrew-jx release has not yet completed

**Comment by garethjevans** [comment]
there is a window where this could happen.  We are moving the homebrew-jx update downstream of jenkins-x-versions (rather than jx) to improve this

**Comment by vfarcic** [comment]
Can we make the release process wait until homebrew-jx update is finished? Otherwise, it is misleading at best.

**Comment by vfarcic** [comment]
I don't think it's a "window". I just run the following.

```bash
NAME               VERSION
jx                 2.0.188
jenkins x platform 2.0.330
Kubernetes cluster v1.13.5-gke.10
kubectl            v1.14.2
helm client        Client: v2.14.0+g05811b8
git                git version 2.20.1 (Apple Git-117)
Operating System   Mac OS X 10.14.5 build 18F203
WARNING:
A new jx version is available: 2.0.192
? Would you like to upgrade to the new jx version? Yes
```

The output is:

```
You are already on the latest version of jx 2.0.188
```

So, it tried to move me 4 releases ahead but it exited thinking that the one four releases ago is the current one.

**Comment by garethjevans** [comment]
The idea is that the initial jx binary release with be marked as a "pre-release", and only when jenkins-x-versions has passed, the jx binary will be marked as verified

**Comment by garethjevans** [comment]
we will need to make some changes to jx to support this

**Comment by garethjevans** [comment]
This should be resolved in 2.0.277
